python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -74,6 +80,7 @@ def __init__(
         cvalue: ctypes.c_void_p,
         numba_type: types.Type,
         value_type: types.Type,
+        iterator_io: IteratorIO,
```

Comment:
Can we update the docstring above to:

- include information about `iterator_io`?
- reflect that `advance` and `dereference` must be properties, and describe how the value of that property is affected by `iterator_io`?